### PR TITLE
Add and remove domains when a project is updated

### PR DESCRIPTION
**Pull request by recurser**
As discussed in [terraform-provider-vercel#28](https://github.com/chronark/terraform-provider-vercel/issues/28)

### CODE REVIEW

File: internal/utility/utility.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package utility
```
Comment (recurser) [inline internal/utility/utility.go:1]: @chronark it'd be simpler to use a 3rd-party library for this, but it didn't seem worth adding an extra dependency for something so small. Feel free to swap this out / move it if you'd like to structure things differently.

Comment (chronark) [reply internal/utility/utility.go:1]: Yeah I agree, no reason to bring in a dependency for this.
I'm thinking about moving it to `/pkg/util` though, what do you think?

Great work, thank you.

**Comment by chronark** [comment]
Thanks so much. this is great.

I'd like to add some tests for this but I won't have time in the upcoming week.
Do you think you can do it? Otherwise I'll do it sometime next week.

Also let me know if you need this released today/tomorrow or if it can wait until next week please.

**Comment by recurser** [comment]
Hi @chronark 

> I'm thinking about moving it to /pkg/util though, what do you think?

Sounds good - done 👍 

> I'd like to add some tests for this

I've tried to add a test for the domain somewhere around [here](https://github.com/chronark/terraform-provider-vercel/blob/eb9fa5c2ec289d9b84615efa69c59f0a57a1a15c/internal/provider/resource_project_test.go#L36-L42) but I haven't been able to get it working. Because it's not a simple property like `InstallCommand` etc, but rather a separate API call during project creation / update, I'm struggling to figure out how to do it. I'm happy to wait until you have time to write them yourself, or try again if you have any pointers.

> let me know if you need this released today/tomorrow or if it can wait until next week please

No rush at all! Whenever suits you is fine by me :)

**Comment by chronark** [comment]
Yeah I am having trouble testing terraform myself.
Maybe this works:
1. Create a project with domain attached
2. Check if the domain is added by calling the project api.
It should be inside the returned struct [here](https://github.com/chronark/terraform-provider-vercel/blob/eb9fa5c2ec289d9b84615efa69c59f0a57a1a15c/pkg/vercel/project/types.go#L59)
though I am not 100% sure

So you would create a new `testAccDomainHasBeenAdded` (or similar) function and call the `Read` method from `pkg/vercel/project` 

I'll be catching up with work this week but I have a few hours in the evening to sort this out together with you.

File: internal/provider/resource_project_test.go [hunk 2]
```diff
@@ -11,6 +11,12 @@ import (
 	"testing"
 )
 
+// The domain to add to the project. The TLD should be authorized in your test Vercel account.
+const domainAlias = "domain-alias.chronark.com"
+
+// The repository to test project creation with. Your Vercel account should have access to this.
+const repository = "chronark/terraform-provider-vercel"
```
Comment (recurser) [inline internal/provider/resource_project_test.go:18]: @chronark I added these constants to make it easier to run it against my own account. Ideally these could be over-ridden by env-vars, but that's probably better done in a separate PR.

**Comment by recurser** [comment]
@chronark thanks - I've added some tests. It might be a bit cleaner to add support for the domain into `testAccCheckProjectStateHasValues` and `testAccCheckActualProjectHasValues`, but it got quite complex when I tried, and I think this is easier to read despite being more verbose. I'm happy to clean things up if you have any concerns / comments.

File: internal/provider/resource_project_test.go (hunk 2 above)
Comment (chronark) [reply internal/provider/resource_project_test.go:18]: Yes good idea. I have created #40 to keep it in mind

**Review by chronark — APPROVED** [review]

**Comment by chronark** [comment]
Great! 
I agree, let's keep it verbose rather than clever and complicated.

I just checked locally and the tests run fine. I still have no idea why it doesn't work on github actions. but that's another issue entirely.

If you are happy with everything, I'd merge it.

**Comment by recurser** [comment]
Thanks @chronark ! `v0.11.0-rc.1` works perfectly for me here both creating and updating domains 👍 

> I still have no idea why it doesn't work on github actions. but that's another issue entirely.

I had the same issue on another project and it took me months to figure it out. Github [changed the behavior of secrets](https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/) so that they aren't available to forks (which makes sense I guess - presumably I could submit a PR with an altered action that sends me your Vercel key somehow). I think if you remade the same PR it would pass.

> If you are happy with everything, I'd merge it.

That'd be great 🎉

**Comment by chronark** [comment]
Ooooh yeah that seems to be exactly the issue, because it works for my own PRs. Thank you

And thanks for your contribution, I'll release v0.11.0 right now
